app/controllers/organizations_controller.rb

```diff
@@ -69,14 +69,24 @@ def deactivate_user
     user = User.with_discarded.find_by!(id: params[:user_id])
     raise ActiveRecord::RecordNotFound unless user.has_role?(Role::ORG_USER, current_organization)
     user.discard!
-    redirect_to user_update_redirect_path, notice: "User has been deactivated."
   end
 
   def reactivate_user
     user = User.with_discarded.find_by!(id: params[:user_id])
     raise ActiveRecord::RecordNotFound unless user.has_role?(Role::ORG_USER, current_organization)
     user.undiscard!
-    redirect_to user_update_redirect_path, notice: "User has been reactivated."
+  end
+
+  def opt_in_email_notification
+    organization = Organization.find(params[:id])
+    organization.update(email_notification_opt_in: true)
+    render json: { message: "Email notifications opted in successfully" }
+  end
+
+  def opt_out_email_notification
```

Comment:
Still unsure why these are their own actions instead of just adding a field to the existing edit page?

Reply:
def organization_params
    request_type_formatter(params)

    params.require(:organization).permit(
       :email_notification_opt_in,
       :email_notification_opt_out
    )
  end


Like That?

Reply:
Just add these 2 fields to the existing `organization_params` method.